extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -789,29 +789,36 @@ export class AmpConsent extends AMP.BaseElement {
     if (!this.isGranularConsentExperimentOn_) {
       return Promise.resolve(true);
     }
-    return this.getPurposeConsentRequired_().then((purposeConsentRequired) => {
-      // True if there are no required purposes
-      if (!purposeConsentRequired || !purposeConsentRequired.length) {
-        return true;
-      }
-      const storedPurposeConsents = consentInfo['purposeConsents'];
-      // False if there are no stored purposes
-      if (
-        !storedPurposeConsents ||
-        Object.keys(storedPurposeConsents).length <
-          purposeConsentRequired.length
-      ) {
-        return false;
-      }
-      // Check if we have a stored consent for each purpose required
-      for (let i = 0; i < purposeConsentRequired.length; i++) {
-        const purpose = purposeConsentRequired[i];
-        if (!hasOwn(storedPurposeConsents, purpose)) {
+    return this.getPurposeConsentRequired_()
+      .then((purposeConsentRequired) => {
+        // True if there are no required purposes
+        if (!purposeConsentRequired || !purposeConsentRequired.length) {
```

Comment:
```suggestion
        if (!purposeConsentRequired?.length) {
```

Reply:
Sweet, getting used to using this operator :)

Reply:
Me too! It's awesome we can finally use it